src/database/firebase.tsx

```diff
@@ -95,6 +95,34 @@ export const addTree = async (tree: Tree) => {
   }
 };
 
+/**
+ * same functionality as addTree but for comments:
+ * assigns uuid to comment and adds to commentCollection.
+ */
+export const saveComment = async (comment: Comment) => {
+  try {
+    const ref = await commentCollection.add(comment);
+    commentCollection.doc(ref.id).update({ uuid: ref.id });
+  } catch (e) {
+    console.warn(e);
+    throw e;
+    // TODO: Add error handling.
+  }
+};
+
+export const addComment = async (comment: Comment, uuid: string) => {
+  try {
+    const ref = await treeCollection.doc(uuid).get();
+    treeCollection
+      .doc(uuid)
+      .update({ comments: [...(ref.data().comments ?? []), comment] });
```

Comment:
yeah this wasn't the best solution my bad, can you actually try this after looking into it --

https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array

```
treeCollection.doc(uuid).update({comments: FieldValue.arrayUnion(comment)})
```

that way you don't need a ref anymore

Note: this might not be the exact code, but it's pretty close